cpp/src/reductions/reductions.cpp

```diff
@@ -68,7 +68,8 @@ struct reduce_dispatch_functor {
       } break;
       case aggregation::MEDIAN: {
         auto sorted_indices = sorted_order(table_view{{col}}, {}, {null_order::AFTER}, stream, mr);
-        auto valid_sorted_indices = split(*sorted_indices, {col.size() - col.null_count()})[0];
+        auto valid_sorted_indices =
+          split(*sorted_indices, {col.size() - col.null_count()}, stream)[0];
         auto col_ptr =
           quantile(col, {0.5}, interpolation::LINEAR, valid_sorted_indices, true, stream, mr);
```

Comment:
Unrelated to this PR change, but should `col_ptr` need to be created using `mr` allocator or default?

Reply:
default, I think? Looks like temporary to me. Also the above `sorted_indices`.

Reply:
The slice/split interface change has made the PR quite a bit more difficult to review atm. I intend to include this fix in a follow up PR.

Reply:
https://github.com/rapidsai/cudf/pull/9344